inc/WebhookHandler/SourceForge.php

```diff
@@ -0,0 +1,92 @@
+<?php
+/**
+ * SourceForge webhook handler class.
+ *
+ * @since 3.0.0
+ *
+ * @package Required\Traduttore
+ */
+
+namespace Required\Traduttore\WebhookHandler;
+
+use Required\Traduttore\ProjectLocator;
+use Required\Traduttore\Repository;
+use Required\Traduttore\Updater;
+use WP_Error;
+use WP_REST_Response;
+
+/**
+ * SourceForge webhook handler class.
+ *
+ * @since 3.0.0
+ *
+ * @see https://forge-allura.apache.org/p/allura/wiki/Webhooks/
+ */
+class SourceForge extends Base {
+	/**
+	 * Permission callback for incoming SourceForge webhooks.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return bool True if permission is granted, false otherwise.
+	 */
+	public function permission_callback(): ?bool {
+		$token   = $this->request->get_header( 'x-allura-signature' );
+		$params  = $this->request->get_params();
+		$locator = new ProjectLocator( $params['repository']['url'] ?? null );
+		$project = $locator->get_project();
+		$secret  = $this->get_secret( $project );
+
+		if ( ! $token ) {
+			return false;
+		}
+
+		$payload_signature = 'sha1=' . hash_hmac( 'sha1', $this->request->get_body(), $secret );
+
+		return hash_equals( $token, $payload_signature );
+	}
+
+	/**
+	 * Callback for incoming SourceForge webhooks.
+	 *
+	 * @since 3.0.0
+	 *
+	 * @return WP_Error|WP_REST_Response REST response on success, error object on failure.
+	 */
+	public function callback() {
+		$params = $this->request->get_params();
+
+		// We only care about the default branch (we have to assume it's master) but don't want to send an error still.
+		if ( 'refs/heads/master' !== $params['ref'] ) {
+			return new WP_REST_Response( [ 'result' => 'Not the default branch' ] );
+		}
```

Comment:
Might be worth making the default branch name filterable / configurable. Might also be relevant for others.